src/components/Sing/ScoreSequencer.vue

```diff
@@ -152,63 +161,86 @@ export default defineComponent({
     const dragMoveCurrentX = ref();
     const dragMoveCurrentY = ref();
     const dragDurationCurrentX = ref();
-    // シーケンサグリッド
-    const gridY = midiKeys;
-    const gridX = computed(() => {
-      const resolution = state.score?.resolution || 480;
-      // NOTE: 最低長: 仮32小節...MIDI長さ(曲長さ)が決まっていないため、無限スクロール化する or 最後尾に足した場合は伸びるようにするなど？
-      const minDuration = resolution * 4 * 32;
-      const lastNote = state.score?.notes.slice(-1)[0];
-      // Score長さ: スコア長もしくは最低長のうち長い方
-      const totalDuration = lastNote
-        ? Math.max(lastNote.position + lastNote.duration, minDuration)
-        : minDuration;
-      // グリッド幅1/16
-      const gridDuration = resolution / 4;
-      // NOTE: いったん最後尾に足した場合は伸びるようにする
-      const gridsMax = Math.ceil(totalDuration / gridDuration) + 16;
-      return [...Array(gridsMax).keys()].map(
-        (gridNum) => gridNum * gridDuration
-      );
-    });
+    // 分解能（Ticks Per Quarter Note）
+    const tpqn = computed(() => state.score?.resolution ?? 480);
     // ノート
-    const notes = computed(() => state.score?.notes);
-    // 表紙
-    const timeSignatures = computed(() => state.score?.timeSignatures);
+    const notes = computed(() => state.score?.notes ?? []);
+    // 拍子
+    const defaultTimeSignature: TimeSignature = {
+      position: 0,
+      beats: 4,
+      beatType: 4,
+    };
+    const timeSignature = computed(() => {
+      return state.score?.timeSignatures?.[0] ?? defaultTimeSignature;
+    });
     // ズーム状態
     const zoomX = computed(() => state.sequencerZoomX);
     const zoomY = computed(() => state.sequencerZoomY);
-    // スナップサイズ
-    const snapSize = computed(() => state.sequencerSnapSize);
-    const snapWidth = computed(() => (snapSize.value / 4) * zoomX.value);
-    // グリッドサイズ
-    const gridWidth = computed(() => sizeX * zoomX.value);
-    const gridHeight = computed(() => sizeY * zoomY.value);
+    // スナップ
+    const snapTicks = computed(() => {
+      return getNoteDuration(state.sequencerSnapType, tpqn.value);
+    });
+    const snapBaseWidth = computed(() => {
+      return tickToBaseX(snapTicks.value, tpqn.value);
+    });
+    const snapWidth = computed(() => {
+      return snapBaseWidth.value * zoomX.value;
+    });
+    // シーケンサグリッド
+    const gridColumnTicks = snapTicks;
+    const gridColumnBaseWidth = snapBaseWidth;
```

Comment:
とりあえずスナップ幅＝グリッド幅としました。

Reply:
ひとまずこうしていることを書いといてあげると誰かの役に立つかも？
```suggestion
    const gridColumnTicks = snapTicks; // ひとまずスナップ幅＝グリッド幅
    const gridColumnBaseWidth = snapBaseWidth;
```

Reply:
コメントを追加しました！